format/spec.md

```diff
@@ -1680,6 +1680,23 @@ Row-level delete changes:
     * These position delete files must be merged into the DV for a data file when one is created
     * Position delete files that contain deletes for more than one data file need to be kept in table metadata until all deletes are replaced by DVs
 
+Row lineage changes:
+
+* Writers must set the table's `next-row-id` and use the existing `next-row-id` as the `first-row-id` to create a new snapshot
+    * When a table is upgraded to v3, `next_row_id` should be initialized to 0
+    * It is recommended to increment `next-row-id` by the total `added_rows_count` and `existing_rows_count` of all manifests assigned a `first_row_id`
+* Writers must assign a `first_row_id` to new data manifests when writing a manifest list
+    * It is recommended to increment `first_row_id` by a manifest's `added_rows_count` and `existing_rows_count`
+* Readers must assign a `first_row_id` for any data file that does not have an assigned value written in a manifest
```

Comment:
I think it may be better to just link to the other sections of the spec. I think lines like this talk about what an implementation needs to do but don't have the details to actually meaningfully do it.